src/yamlProjectConfiguration/models.ts

```diff
@@ -13,3 +13,9 @@ export enum TriggerType {
     cron = "cron",
     http = "http",
 }
+
+export enum FunctionProviderType {
+    aws = "aws",
+    azure = "azure",
+    gcp = "gcp",
```

Comment:
Maybe we should comment these for now as they are not implemented yet. You can also leave a TODO comment.